rpmlb/recipe.py

```diff
@@ -64,3 +83,54 @@ def verify(self):
             assert(package_dict['name'])
 
         return True
+
+    @staticmethod
+    def _package_name(package: Union[str, dict]) -> str:
+        """Extract package name.
+
+        Keyword arguments:
+            package: Raw package data to analyze.
+                It should be either string or one-item dictionary.
+
+        Returns:
+            Name of the package.
+
+        Raises:
+            ValueError: Invalid package type.
+        """
+
+        if isinstance(package, str):  # The name itself
+            return package
+
+        elif isinstance(package, dict):
+            # package should be one-item dictionary
+            if len(package) != 1:
+                msg = 'Invalid package data: {}'.format(package)
+                raise ValueError(msg)
+
+            # The name is the only key
+            name, = package.keys()
+            return name
+
+        else:
+            msg = 'Invalid type of package: {!r}'.format(type(package))
+            raise ValueError(msg)
+
+    def _count_bootstrap_sequences(self) -> Mapping[str, Iterator[int]]:
+        """Count package instances and produce bootstrap sequences.
+
+        Returns:
+            Mapping of package number to iterator of bootstrap numbers.
+        """
+
+        name_list = map(self._package_name, self.recipe['packages'])
+        count_map = Counter(name_list)
+
+        # For each package count it's instances except the last one;
+        # start at 1
+        sequence_map = {
+            name: iter(range(1, count))
+            for name, count in count_map.items()
```

Comment:
About your code.

```
name: iter(range(1, count))
for name, count in count_map.items()
```

Below 2 cases looks better than it?


```
name: iter(range(1, count))
    for name, count in count_map.items()
```

```
name: iter(range(1, count)) for name, count in count_map.items()
```

Or this code is including several statements per 1 line.
It is different from below document's policy.
Is it OK?

http://python-guide-pt-br.readthedocs.io/en/latest/writing/style/
> One statement per line